### PR TITLE
Enabling GenSspiClientContext() to be used in Windows

### PR DESCRIPTION
**Pull request by geleems**
This change enables GenSspiClientContext() to be used in Windows.

### CODE REVIEW

**Comment by saurabh500** [comment]
Do you plan to add the Csproj changes with this as well ?

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs [hunk 1]
```diff
@@ -143,15 +143,21 @@ public void GenSspiClientContext(SspiClientContextStatus sspiClientContextStatus
             }
 
             sendBuff = outSecurityBuffer.token;
+            if (sendBuff == null)
+            {
+                sendBuff = new byte[0];
```
Comment (justinvp) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs]: `Array.Empty<byte>()`?

File: src/System.Data.SqlClient/src/Resources/Strings.resx [hunk 2]
```diff
@@ -1086,4 +1086,25 @@
...
+  </data>
+  <data name="event_OperationReturnedSomething" xml:space="preserve">
+    <value>{0} returned {1}.</value>
+  </data>
+  <data name="net_invalid_enum" xml:space="preserve">
+    <value>The specified value is not valid in the '{0}' enumeration.</value>
+  </data>
+  <data name="SSPIInvalidHandleType" xml:space="preserve">
+    <value>'{0}' is not a supported handle type.</value>
+  </data>
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/Resources/Strings.resx]: Why are these being added?  They're not being consumed elsewhere in this PR.

**Comment by saurabh500** [comment]
@stephentoub These changes are part of the work being done to get the NegotiateStreamPal for Windows into System.Data.SqlClient so that Managed SNI can do Windows Integrate auth as well. 
Gene is preparing the csproj file as well. It may come in as a separate PR or as part of this PR. 

The strings are needed for NegotiateStreamPal for Windows.

**Comment by stephentoub** [comment]
> These changes are part of the work being done to get the NegotiateStreamPal for Windows into System.Data.SqlClient so that Managed SNI can do Windows Integrate auth as well

Thanks, @saurabh500.  Am I understanding correctly then that these need to be added to enable it to build for Windows, but this change doesn't include the project changes to get it building for Windows because we're not yet ready for that?  If so, understood, thanks.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs (hunk 1 above)
Comment (geleems) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs]: fixed.

**Comment by geleems** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Pull request merged**